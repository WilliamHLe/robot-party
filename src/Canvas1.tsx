$(document).ready(function() {
    //Endrer SVG elementenes farge når man trykker på de
    let c = document.getElementById("canvas");
    let ctx = c.getContext("2d");
    //Data for alle figurene som skal tegner i canvas
    let dance = [
        //Head
        { x: 320, y: 175, w: 105, h: 125, rotate: 0 },
        //Left arm
        { x: 365, y: 290, w: 50, h: 100, rotate: -10 },
        {
            x: 385,
            y: 355,
            w: 100,
            h: 50,
            rotate: 10
        },
        //Left foot
        { x: 400, y: 400, w: 60, h: 125, rotate: -40 },
        { x: 425, y: 490, w: 50, h: 100, rotate: 10 },
        //Body
        { x: 375, y: 325, w: 0, h: 0, rotate: 0 },
        { x: 375, y: 375, w: 0, h: 0, rotate: 0 },
        //Right arm
        { x: 345, y: 300, w: 50, h: 100, rotate: -10 },
        {
            x: 365,
            y: 365,
            w: 100,
            h: 50,
            rotate: 10
        },
        //Right foot
        { x: 325, y: 415, w: 60, h: 125, rotate: 5 },
        { x: 315, y: 515, w: 50, h: 100, rotate: 20 }
    ];
    let bop = [
        //Head
        { x1: 375, x2: 450, x3: 300, y1: 275, y2: 150 },
        //Left arm
        { x: 450, y: 280, w: 50, h: 100, rotate: 90 },
        {
            x: 475,
            y: 275,
            w: 100,
            h: 50,
            rotate: -80
        },
        //Right arm
        { x: 255, y: 280, w: 50, h: 100, rotate: 90 },
        {
            x: 175,
            y: 275,
            w: 100,
            h: 50,
            rotate: 80
        },
        //Right foot
        { x: 290, y: 400, w: 60, h: 125, rotate: 40 },
        { x: 265, y: 490, w: 50, h: 100, rotate: -10 },
        //Left foot
        { x: 410, y: 400, w: 60, h: 125, rotate: -40 },
        { x: 435, y: 490, w: 50, h: 100, rotate: 10 },
        //Body
        { x: 375, y: 325, w: 0, h: 0, rotate: 0 },
        { x: 375, y: 375, w: 0, h: 0, rotate: 0 }

    ];
    let circles = [
        {x:375,y:275,r:150,sA:0.75*Math.PI,eA:0.25*Math.PI,color:"rgb(255,0,0)"},
        {x:375,y:450,r:200,sA:0,eA:2*Math.PI,color:"rgb(0,255,0)"},
        {x:450,y:475,r:75,sA:0,eA:2*Math.PI,color:"rgb(0,0,255)"},
        {x:375,y:75,r:25,sA:0,eA:2*Math.PI,color:"rgb(0,0,255)"}

    ]
    drawFloor();
    ctx.fillStyle = "yellow";
    ctx.lineWidth = 5;
    //Tegner figurene
    function drawDancer(arr) {
        arr.forEach(r => {
            ctx.save();
            ctx.beginPath();
            ctx.fillStyle = "yellow";
            if (r.w == 0) {
                ctx.arc(r.x, r.y, 55, 0, 2 * Math.PI);
            } else if (typeof r.x1 !== "undefined") {
                ctx.beginPath();
                ctx.moveTo(r.x1, r.y1);
                ctx.lineTo(r.x2, r.y2);
                ctx.lineTo(r.x3, r.y2);
                ctx.closePath();
                ctx.stroke();
                ctx.fill();
            } else {
                //Kode som gjør av rektangler roterer rundt midtpunket sitt. Koden er funnet her: https://stackoverflow.com/questions/17125632/html5-canvas-rotate-object-without-moving-coordinates/17126036
                ctx.translate(r.x + r.w / 2, r.y + r.h / 2);
                ctx.rotate((r.rotate * Math.PI) / 180);
                ctx.rect(-r.w / 2, -r.h / 2, r.w, r.h);
            }
            ctx.fill();
            ctx.stroke();
            ctx.restore();
        });
    }
    function drawCircles() {
        ctx.beginPath();
        ctx.moveTo(375, 125);
        ctx.lineTo(375, 100);
        ctx.stroke();
        circles.forEach(c => {
            ctx.beginPath();
            ctx.fillStyle = c.color;
            ctx.arc(c.x,c.y,c.r,c.sA,c.eA);
            ctx.fill();
            ctx.stroke();
        })
    }
    //Tegner gulvet
    function drawFloor() {
        ctx.globalAlpha = 1;
        let gradient = ctx.createLinearGradient(0, 450, 0, 750);
        gradient.addColorStop(0, "grey");
        gradient.addColorStop(1, "white");
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 450, 750, 300);
        //ctx.globalAlpha = 0.5;
    }

    let j = 1;
    let forward = true;
    let speed = 30;
    let utstilling = "bop";
    requestAnimationFrame(animate);
    function animate(){
        ctx.clearRect(0, 0, c.width, c.height);
        drawFloor();
        if(utstilling == "dance") {
            drawDancer(dance);
        } else if(utstilling == "blink"){
            drawCircles();
            if(j > speed || j < 1) {
                circles.forEach(c => {
                    let r = Math.floor(Math.random() * 255);
                    let g = Math.floor(Math.random() * 255);
                    let b = Math.floor(Math.random() * 255);
                    c.color = "rgb("+r+","+g+","+b+")";
                })
            }
        } else if(utstilling == "bop") {
            drawDancer(bop);
        }
        if(forward){
            //Utstilling 1
            dance[2].x += 40/speed;
            dance[2].y -= 95/speed;
            dance[2].rotate -= 90/speed;
            dance[1].x += 35/speed;
            dance[1].y -= 15/speed;
            dance[1].rotate -= 75/speed;
            dance[8].x += 50/speed;
            dance[8].y -= 60/speed;
            dance[8].rotate -= 10/speed;
            dance[7].x += 20/speed;
            dance[7].y -= 25/speed;
            dance[7].rotate -= 70/speed;
            //Utstilling 2
            bop[0].y1 += 50/speed;
            bop[0].y2 += 50/speed;
            bop[1].y += 50/speed;
            bop[2].y += 50/speed;
            bop[3].y += 50/speed;
            bop[4].y += 50/speed;
            bop[9].y += 50/speed;
            bop[10].y += 50/speed;
            bop[6].rotate -= 20/speed;
            bop[8].rotate += 20/speed;
            bop[5].rotate += 20/speed;
            bop[7].rotate -= 20/speed;
            j++;
        } else {
            //Utstilling 1
            dance[2].x -= 40/speed;
            dance[2].y += 95/speed;
            dance[2].rotate += 90/speed;
            dance[1].x -= 35/speed;
            dance[1].y += 15/speed;
            dance[1].rotate += 75/speed;
            dance[8].x -= 50/speed;
            dance[8].y += 60/speed;
            dance[8].rotate += 10/speed;
            dance[7].x -= 20/speed;
            dance[7].y += 25/speed;
            dance[7].rotate += 70/speed;
            //Utstilling 2
            bop[0].y1 -= 50/speed;
            bop[0].y2 -= 50/speed;
            bop[1].y -= 50/speed;
            bop[2].y -= 50/speed;
            bop[3].y -= 50/speed;
            bop[4].y -= 50/speed;
            bop[9].y -= 50/speed;
            bop[10].y -= 50/speed;
            bop[6].rotate += 20/speed;
            bop[8].rotate -= 20/speed;
            bop[5].rotate -= 20/speed;
            bop[7].rotate += 20/speed;
            j--;
        }
        if(j > speed){
            forward = false;
        } else if (j<1) {
            forward = true;
        }
        requestAnimationFrame(animate);
    }
});

