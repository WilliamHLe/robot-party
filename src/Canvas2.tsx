let j = 1;
let forward = true;
let speed = 30;
let utstilling = "bop";
let c = document.getElementById("canvas");
let ctx = c.getContext("2d");
let bop = [
    //Head
    { x1: 375, x2: 450, x3: 300, y1: 275, y2: 150 },
    //Left arm
    { x: 450, y: 280, w: 50, h: 100, rotate: 90 },
    {
        x: 475,
        y: 275,
        w: 100,
        h: 50,
        rotate: -80
    },
    //Right arm
    { x: 255, y: 280, w: 50, h: 100, rotate: 90 },
    {
        x: 175,
        y: 275,
        w: 100,
        h: 50,
        rotate: 80
    },
    //Right foot
    { x: 290, y: 400, w: 60, h: 125, rotate: 40 },
    { x: 265, y: 490, w: 50, h: 100, rotate: -10 },
    //Left foot
    { x: 410, y: 400, w: 60, h: 125, rotate: -40 },
    { x: 435, y: 490, w: 50, h: 100, rotate: 10 },
    //Body
    { x: 375, y: 325, w: 0, h: 0, rotate: 0 },
    { x: 375, y: 375, w: 0, h: 0, rotate: 0 }

];
function drawDancer(arr) {
    arr.forEach(r => {
        ctx.save();
        ctx.beginPath();
        ctx.fillStyle = "yellow";
        if (r.w == 0) {
            ctx.arc(r.x, r.y, 55, 0, 2 * Math.PI);
        } else if (typeof r.x1 !== "undefined") {
            ctx.beginPath();
            ctx.moveTo(r.x1, r.y1);
            ctx.lineTo(r.x2, r.y2);
            ctx.lineTo(r.x3, r.y2);
            ctx.closePath();
            ctx.stroke();
            ctx.fill();
        } else {
            //Kode som gjÃ¸r av rektangler roterer rundt midtpunket sitt. Koden er funnet her: https://stackoverflow.com/questions/17125632/html5-canvas-rotate-object-without-moving-coordinates/17126036
            ctx.translate(r.x + r.w / 2, r.y + r.h / 2);
            ctx.rotate((r.rotate * Math.PI) / 180);
            ctx.rect(-r.w / 2, -r.h / 2, r.w, r.h);
        }
        ctx.fill();
        ctx.stroke();
        ctx.restore();
    });
}
//Tegner gulvet
function drawFloor() {
    ctx.globalAlpha = 1;
    let gradient = ctx.createLinearGradient(0, 450, 0, 750);
    gradient.addColorStop(0, "grey");
    gradient.addColorStop(1, "white");
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 450, 750, 300);
    //ctx.globalAlpha = 0.5;
}
function animate(){
    ctx.clearRect(0, 0, c.width, c.height);
    drawFloor();
    if(utstilling == "dance") {
        drawDancer(dance);
    } else if(utstilling == "blink"){
        drawCircles();
        if(j > speed || j < 1) {
            circles.forEach(c => {
                let r = Math.floor(Math.random() * 255);
                let g = Math.floor(Math.random() * 255);
                let b = Math.floor(Math.random() * 255);
                c.color = "rgb("+r+","+g+","+b+")";
            })
        }
    } else if(utstilling == "bop") {
        drawDancer(bop);
    }
    if(forward){
        bop[0].y1 += 50/speed;
        bop[0].y2 += 50/speed;
        bop[1].y += 50/speed;
        bop[2].y += 50/speed;
        bop[3].y += 50/speed;
        bop[4].y += 50/speed;
        bop[9].y += 50/speed;
        bop[10].y += 50/speed;
        bop[6].rotate -= 20/speed;
        bop[8].rotate += 20/speed;
        bop[5].rotate += 20/speed;
        bop[7].rotate -= 20/speed;
        j++;
    } else {
        bop[0].y1 -= 50/speed;
        bop[0].y2 -= 50/speed;
        bop[1].y -= 50/speed;
        bop[2].y -= 50/speed;
        bop[3].y -= 50/speed;
        bop[4].y -= 50/speed;
        bop[9].y -= 50/speed;
        bop[10].y -= 50/speed;
        bop[6].rotate += 20/speed;
        bop[8].rotate -= 20/speed;
        bop[5].rotate -= 20/speed;
        bop[7].rotate += 20/speed;
        j--;
    }
    if(j > speed){
        forward = false;
    } else if (j<1) {
        forward = true;
    }
    requestAnimationFrame(animate);
}