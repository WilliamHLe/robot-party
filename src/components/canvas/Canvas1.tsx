import React, { useRef, useEffect} from 'react'
import "../container.css"

interface CanvasProps {
    speedValue: number;
    colorValue: string;
}

const Canvas1 = ({speedValue, colorValue}: CanvasProps, props: JSX.IntrinsicAttributes & React.ClassAttributes<HTMLCanvasElement> & React.CanvasHTMLAttributes<HTMLCanvasElement>) => {

    const canvasRef = useRef(null)

    let j = 1;
    let forward = true;
    //speedValue oppdateres til riktig verdi, men vises ikke i bildet?
    //problem med render?
    let speed = speedValue;
    //let utstilling = "bop";
    let dance = [
        //Head
        { x: 320, y: 175, w: 105, h: 125, rotate: 0 },
        //Left arm
        { x: 365, y: 290, w: 50, h: 100, rotate: -10 },
        {
            x: 385,
            y: 355,
            w: 100,
            h: 50,
            rotate: 10
        },
        //Left foot
        { x: 400, y: 400, w: 60, h: 125, rotate: -40 },
        { x: 425, y: 490, w: 50, h: 100, rotate: 10 },
        //Body
        { x: 375, y: 325, w: 0, h: 0, rotate: 0 },
        { x: 375, y: 375, w: 0, h: 0, rotate: 0 },
        //Right arm
        { x: 345, y: 300, w: 50, h: 100, rotate: -10 },
        {
            x: 365,
            y: 365,
            w: 100,
            h: 50,
            rotate: 10
        },
        //Right foot
        { x: 325, y: 415, w: 60, h: 125, rotate: 5 },
        { x: 315, y: 515, w: 50, h: 100, rotate: 20 }
    ];

    const drawFloor = (ctx:any) => {
        ctx.globalAlpha = 1;
        let gradient = ctx.createLinearGradient(0, 450, 0, 750);
        gradient.addColorStop(0, "grey");
        gradient.addColorStop(1, "white");
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 0, 800, 800);
        //ctx.globalAlpha = 0.5;
    }

    const drawDancer = (arr:any, ctx:any) => {
        arr.forEach((r: { w: number; x: number; y: number; x1: any; y1: any; x2: any; y2: any; x3: any; h: number; rotate: number; }) => {
            ctx.save();
            ctx.beginPath();
            ctx.fillStyle = "yellow";
            if (r.w === 0) {
                ctx.arc(r.x, r.y, 55, 0, 2 * Math.PI);
            } else if (typeof r.x1 !== "undefined") {
                ctx.beginPath();
                ctx.moveTo(r.x1, r.y1);
                ctx.lineTo(r.x2, r.y2);
                ctx.lineTo(r.x3, r.y2);
                ctx.closePath();
                ctx.stroke();
                ctx.fill();
            } else {
                //Kode som gjÃ¸r av rektangler roterer rundt midtpunket sitt. Koden er funnet her: https://stackoverflow.com/questions/17125632/html5-canvas-rotate-object-without-moving-coordinates/17126036
                ctx.translate(r.x + r.w / 2, r.y + r.h / 2);
                ctx.rotate((r.rotate * Math.PI) / 180);
                ctx.rect(-r.w / 2, -r.h / 2, r.w, r.h);
            }
            ctx.fill();
            ctx.stroke();
            ctx.restore();
        });
    }

    useEffect(() => {
        const canvas:any = canvasRef.current
        const context = canvas.getContext('2d')

        const animate = () => {
            context.clearRect(0, 0, 800, 800);
            drawFloor(context)
            drawDancer(dance, context);
            if(forward){
                dance[2].x += 40/speed;
                dance[2].y -= 95/speed;
                dance[2].rotate -= 90/speed;
                dance[1].x += 35/speed;
                dance[1].y -= 15/speed;
                dance[1].rotate -= 75/speed;
                dance[8].x += 50/speed;
                dance[8].y -= 60/speed;
                dance[8].rotate -= 10/speed;
                dance[7].x += 20/speed;
                dance[7].y -= 25/speed;
                dance[7].rotate -= 70/speed;
                j++;
            } else {
                dance[2].x -= 40/speed;
                dance[2].y += 95/speed;
                dance[2].rotate += 90/speed;
                dance[1].x -= 35/speed;
                dance[1].y += 15/speed;
                dance[1].rotate += 75/speed;
                dance[8].x -= 50/speed;
                dance[8].y += 60/speed;
                dance[8].rotate += 10/speed;
                dance[7].x -= 20/speed;
                dance[7].y += 25/speed;
                dance[7].rotate += 70/speed;
                j--;
            }
            if(j > speed){
                forward = false;
            } else if (j<1) {
                forward = true;
            }
            requestAnimationFrame(animate);
        }
        requestAnimationFrame(animate);
    }, [])

    return <canvas ref={canvasRef} width={800} height={800} className="canvas"  {...props}/>
}

export default Canvas1