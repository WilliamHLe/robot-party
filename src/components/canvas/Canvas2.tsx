import React, {useRef, useEffect, useContext} from 'react'
import RadioContext from "../context/RadioContext";


const Canvas2 = (props: JSX.IntrinsicAttributes & React.ClassAttributes<HTMLCanvasElement> & React.CanvasHTMLAttributes<HTMLCanvasElement>) => {

    const canvasRef = useRef(null)
    const {speed, color} = useContext(RadioContext);

    let j = 1;
    let forward = true;
    let utstilling = "bop";

    let bop: any = [
        //Head
        { x1: 375, x2: 450, x3: 300, y1: 275, y2: 150 },
        //Left arm
        { x: 450, y: 280, w: 50, h: 100, rotate: 90 },
        {
            x: 475,
            y: 275,
            w: 100,
            h: 50,
            rotate: -80
        },
        //Right arm
        { x: 255, y: 280, w: 50, h: 100, rotate: 90 },
        {
            x: 175,
            y: 275,
            w: 100,
            h: 50,
            rotate: 80
        },
        //Right foot
        { x: 290, y: 400, w: 60, h: 125, rotate: 40 },
        { x: 265, y: 490, w: 50, h: 100, rotate: -10 },
        //Left foot
        { x: 410, y: 400, w: 60, h: 125, rotate: -40 },
        { x: 435, y: 490, w: 50, h: 100, rotate: 10 },
        //Body
        { x: 375, y: 325, w: 0, h: 0, rotate: 0 },
        { x: 375, y: 375, w: 0, h: 0, rotate: 0 }

    ];

    const drawFloor = (ctx:any) => {
        ctx.globalAlpha = 1;
        let gradient = ctx.createLinearGradient(0, 450, 0, 750);
        gradient.addColorStop(0, "grey");
        gradient.addColorStop(1, "white");
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 0, 800, 800);
        //ctx.globalAlpha = 0.5;
    }

    const drawDancer = (arr:any, ctx:any) => {
        arr.forEach((r: { w: number; x: number; y: number; x1: any; y1: any; x2: any; y2: any; x3: any; h: number; rotate: number; }) => {
            ctx.save();
            ctx.beginPath();
            ctx.fillStyle = color;
            if (r.w === 0) {
                ctx.arc(r.x, r.y, 55, 0, 2 * Math.PI);
            } else if (typeof r.x1 !== "undefined") {
                ctx.beginPath();
                ctx.moveTo(r.x1, r.y1);
                ctx.lineTo(r.x2, r.y2);
                ctx.lineTo(r.x3, r.y2);
                ctx.closePath();
                ctx.stroke();
                ctx.fill();
            } else {
                //Kode som gjÃ¸r av rektangler roterer rundt midtpunket sitt. Koden er funnet her: https://stackoverflow.com/questions/17125632/html5-canvas-rotate-object-without-moving-coordinates/17126036
                ctx.translate(r.x + r.w / 2, r.y + r.h / 2);
                ctx.rotate((r.rotate * Math.PI) / 180);
                ctx.rect(-r.w / 2, -r.h / 2, r.w, r.h);
            }
            ctx.fill();
            ctx.stroke();
            ctx.restore();
        });
    }

    useEffect(() => {
        const canvas:any = canvasRef.current
        const context = canvas.getContext('2d')

        const animate = () => {
            context.clearRect(0, 0, 800, 800);
            drawFloor(context);
            if(utstilling === "dance") {

            } else if(utstilling === "blink"){

            } else if(utstilling === "bop") {
                drawDancer(bop, context);
            }
            if(forward){
                bop[0].y1 += 50/speed;
                bop[0].y2 += 50/speed;
                bop[1].y += 50/speed;
                bop[2].y += 50/speed;
                bop[3].y += 50/speed;
                bop[4].y += 50/speed;
                bop[9].y += 50/speed;
                bop[10].y += 50/speed;
                bop[6].rotate -= 20/speed;
                bop[8].rotate += 20/speed;
                bop[5].rotate += 20/speed;
                bop[7].rotate -= 20/speed;
                j++;
            } else {
                bop[0].y1 -= 50/speed;
                bop[0].y2 -= 50/speed;
                bop[1].y -= 50/speed;
                bop[2].y -= 50/speed;
                bop[3].y -= 50/speed;
                bop[4].y -= 50/speed;
                bop[9].y -= 50/speed;
                bop[10].y -= 50/speed;
                bop[6].rotate += 20/speed;
                bop[8].rotate -= 20/speed;
                bop[5].rotate -= 20/speed;
                bop[7].rotate += 20/speed;
                j--;
            }
            if(j > speed){
                forward = false;
            } else if (j<1) {
                forward = true;
            }
            requestAnimationFrame(animate);
        }
        requestAnimationFrame(animate);
    }, [speed, color])

    return <canvas ref={canvasRef} width={800} height={800} {...props}/>
}

export default Canvas2

