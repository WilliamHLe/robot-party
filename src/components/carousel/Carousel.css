.carousel-wrapper {
    display: flex;
    flex-flow: column;
    align-items: center;
    padding: 30px 60px 60px 30px;
}
.carousel-slides {
    transition: all 1s ease;
    /*sliter med å få transition siden komponenten rerenderer?*/
}
.slide-active {
    display: block;
    width: 100%;
}
.slide-hidden {
    display: none;
    width: 100%;
}
.carousel-buttons {
    display: inline-block;
    width: 90%;
}
.carousel-button {
    background-color: #0D1A2B;
    color: white;
    cursor: pointer;
    padding: 17px 35px;
    text-align: center;
    font-size: 18px;
    border-radius: 8px;
    border: none;
}
.carousel-button:hover {
    background-color: #1c375c;
}

