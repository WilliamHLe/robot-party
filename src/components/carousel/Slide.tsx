import React from "react";
import DisplayPoetry from "../DisplayPoetry";
import "../container.css"
import Music from "../Music";


interface SlideProps {
    url: string;
    canvas: JSX.Element;
}

//Slide returnerer innholdet selve <div>en som skal vises i slideshowet
//Tar inn en url for Ã¥ hente riktig dikt og en canvas-komponent som props
const Slide = ({url, canvas}: SlideProps) => {
    return (
        <div className="grid-container">
            <div className="grid-canvas">
                {canvas}
            </div>
            <div className="grid-poetry">
                <DisplayPoetry url={url} />
            </div>
            <div className="grid-audio">
                <Music />
            </div>

        </div>
    )
}

export default Slide;