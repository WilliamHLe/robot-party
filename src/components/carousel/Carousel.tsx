import React, {useState} from "react";
import "./Carousel.css";

/* Based on: https://medium.com/octopus-wealth/creating-a-carousel-in-react-e45918738212 */

/*Forenklet kode fra tutorial gitt i link over (blant annet tatt bort bruk av Styled Components).*/

/*Denne koden ga et godt utgangspunkt ettersom den lot slides være egne komponenter, som jeg ønsket for å gjøre
* komponenten mer generell.*/

interface IProps {
    children: JSX.Element[];
}


const Carousel = ({ children }: IProps) => {
    const [currentSlide, setCurrentSlide] = useState<number>(0);

    const activeSlide = children.map((slide, index) => (

        <div className={currentSlide===index ? 'slide-active' : 'slide-hidden'} key={index}>
            {slide}
        </div>
    ));

    return (
        <div>
            <div className="carousel-wrapper">
                <div className="carousel-slides">
                    {activeSlide}
                </div>

                <div className="carousel-buttons">

                    <button className="carousel-button" style={{float: "left"}}
                        onClick={() => {
                            setCurrentSlide((currentSlide - 1 + activeSlide.length) % activeSlide.length);
                        }}
                    >
                        &#8810; Forrige
                    </button>

                    <button className="carousel-button" style={{float: "right"}}
                        onClick={() => {
                            setCurrentSlide((currentSlide + 1) % activeSlide.length);
                        }}
                    >
                        Neste &#8811;
                    </button>

                </div>
            </div>
        </div>
    );
};

export default Carousel;