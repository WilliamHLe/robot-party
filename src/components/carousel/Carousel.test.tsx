import React from 'react';
import Carousel from "./Carousel";
import Slide from "./Slide";
import Canvas1 from "../canvas/Canvas1";
import Canvas2 from "../canvas/Canvas2";

const renderer = require("react-test-renderer");

//carousel must have minimum two children
it('renders correctly with two children that takes url and canvas as props', () => {
    const tree = renderer.create(
        <Carousel>
            <Slide url = {"https://poetrydb.org/author,title/Emily%20Dickinson;We%20should%20not%20mind%20so%20small%20a%20flower"} canvas = {<Canvas1/>}/>
            <Slide url = {"https://poetrydb.org/author,title/Emily%20Dickinson;Could%20Hope%20inspect%20her%20Basis"} canvas = {<Canvas2/>}/>
        </Carousel>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});