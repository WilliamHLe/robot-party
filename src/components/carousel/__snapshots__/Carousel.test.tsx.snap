// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with two children that takes url and canvas as props 1`] = `
<div>
  <div
    className="carousel-wrapper"
  >
    <div
      className="carousel-slides"
    >
      <div
        className="slide-active"
      >
        <div
          className="grid-container"
        >
          <div
            className="grid-canvas"
          >
            <canvas
              className="canvas"
              height={800}
              width={800}
            />
          </div>
          <div
            className="grid-poetry"
          >
            <h1>
              Vennligst vent mens diktet hentes...
            </h1>
          </div>
        </div>
      </div>
      <div
        className="slide-hidden"
      >
        <div
          className="grid-container"
        >
          <div
            className="grid-canvas"
          >
            <canvas
              className="canvas"
              height={800}
              width={800}
            />
          </div>
          <div
            className="grid-poetry"
          >
            <h1>
              Vennligst vent mens diktet hentes...
            </h1>
          </div>
        </div>
      </div>
    </div>
    <div
      className="carousel-buttons"
    >
      <button
        className="carousel-button"
        onClick={[Function]}
        style={
          Object {
            "float": "left",
          }
        }
      >
        ≪ Forrige
      </button>
      <button
        className="carousel-button"
        onClick={[Function]}
        style={
          Object {
            "float": "right",
          }
        }
      >
        Neste ≫
      </button>
    </div>
  </div>
</div>
`;
